server/channels/api4/team.go

```diff
@@ -877,14 +883,21 @@ func addTeamMembers(c *Context, w http.ResponseWriter, r *http.Request) {
 			return
 		}
 
+		// if user cannot invite guests, check if any users are guest users.
+		if !canInviteGuests {
+			user, err := c.App.GetUser(member.UserId)
+			if err != nil || user == nil {
```

Comment:
There isn't a case where err is nil and user will also be nil. So let's just check for `err != nil` and also use `.Wrap()` later to preserve the original error in the logs.

Reply:
done.